Feature/jms implementation

jbarbaso opened the pull request:
Added JsonB API dependencies to pom file.
Changed Timestamp fields to LocalDateTime type in entity Car and files affected.
Added message producer and message listener to send and read messages from "jms/CarsAppQueue".
Added a new resource to send messages to the queue with REST

jbarbaso committed:
Formatted pom.xml file

jbarbaso committed:
Added JsonB API dependencies to pom file

jbarbaso committed:
Changed Timestamp fields to LocalDateTime type in entity Car and files affected.
Added LocalDateTimeAdapter to parse Jax-RS requests fields from text LocalDateTime.
Added @XmlJavaTypeAdapter(LocalDateTimeAdapter.class) to LocalDateTime fields.

jbarbaso committed:
Added message producer and message listener to send and read messages from "jms/CarsAppQueue"

jbarbaso committed:
Added a new resource to send messages to the queue with REST

javieraviles commented on src/main/java/com/everis/cars/entity/LocalDateTimeAdapter.java:
```diff
@@ -0,0 +1,19 @@
+import java.time.format.DateTimeFormatter;
+
+import javax.xml.bind.annotation.adapters.XmlAdapter;
+
```
would be nice to document whats the end goal of having this class

javieraviles commented on src/main/java/com/everis/cars/jms/CreateCarMessageListener.java:
```diff
@@ -0,0 +1,87 @@
+ * @see javax.ejb.MessageDriven
+ */
+@MessageDriven(mappedName = "jms/CarsAppQueue")
+public class CreateCarMessageListener implements MessageListener {
```
is there a unit test for this class?

jbarbaso replied:
I refactorized the class and created the test

javieraviles commented on src/main/java/com/everis/cars/jms/CreateCarMessageProducer.java:
```diff
@@ -0,0 +1,65 @@
+ * Producer helper class to connect and send messages to a queue
+ */
+@Stateless
+public class CreateCarMessageProducer {
```
is there a unit test for this class?

jbarbaso replied:
Unit test created

javieraviles reviewed (changes requested):
Looks neat

jbarbaso committed:
Added javadocs to LocalDateTimeAdapter class

jbarbaso committed:
Rename CreateCarMessageListener to CarMessageListener
Rename CreateCarMessageProducer to Car Message Producer

jbarbaso committed:
Remove unnecessary dependencies

jbarbaso committed:
Added InvalidActionException and ValidatorUtil.
Implemented the validation in CarMessageListener and added the InvalidActionException catch to the onMessage method.

jbarbaso committed:
Remove unnecessary dependencies

jbarbaso committed:
Rename references

jbarbaso committed:
Rename references

jbarbaso committed:
Tests created.
Work in progress

Commit:
Added final when is needed

Commit:
Rename InvalidActionException to InvalidActionMessageException

Commit:
Created InvalidCarMessageException

Commit:
Refactor class

Commit:
Added more tests and comments to CarMessageListenerTest
